hadoop-hdds/server-scm/src/test/java/org/apache/hadoop/hdds/scm/node/TestSCMNodeManager.java

```diff
@@ -170,6 +176,33 @@ public void testScmHeartbeat()
     }
   }
 
+  /**
+   * Tests that Node manager handles Layout versions correctly.
+   *
+   * @throws IOException
+   * @throws InterruptedException
+   * @throws TimeoutException
+   */
+  @Test
+  public void testScmLayoutOnRegister()
+      throws IOException, InterruptedException, AuthenticationException {
+
+    try (SCMNodeManager nodeManager = createNodeManager(getConf())) {
+      LayoutVersionProto layoutInfoSuccess = LayoutVersionProto.newBuilder()
+          .setMetadataLayoutVersion(1).setSoftwareLayoutVersion(1).build();
```

Comment:
Can we refactor this test to use last layout feature's version and last layout feature's version + 1 for success and failure cases? That way, the test does not need change every time a Layout feature is added.

Reply:
Yup, I will do this.